controllers/watermarkpodautoscaler_controller.go

```diff
@@ -696,31 +693,30 @@ func (r *WatermarkPodAutoscalerReconciler) SetupWithManager(mgr ctrl.Manager) er
 	b := ctrl.NewControllerManagedBy(mgr).
 		For(&datadoghqv1alpha1.WatermarkPodAutoscaler{}, builder.WithPredicates(predicate.Funcs{UpdateFunc: updatePredicate}))
 	err := b.Complete(r)
-
 	if err != nil {
 		return err
 	}
 
-	config := mgr.GetConfig()
+	podConfig := rest.CopyConfig(mgr.GetConfig())
```

Comment:
Any ways you could add a comment just for the future so we know we got bitten by that?

Reply:
Indeed, could be useful, added.